informer/src/handler/rpc_ws_handler.rs

```diff
@@ -0,0 +1,103 @@
+    WsError, WsServer,
+};
+use crossbeam::Sender;
+use crossbeam_channel as crossbeam;
+use futures::Future;
+use jsonrpc_core::{futures, BoxFuture};
+use std::io;
+use std::sync::Arc;
+
+pub struct InformerConfig {
+    pub interface: String,
+    pub port: u16,
+    pub max_connections: usize,
+}
+
+#[derive(Default)]
+pub struct Handler {
+    pub handler: PubSubHandler<Arc<Session>>,
+}
+
+impl Handler {
+    pub fn new(handler: PubSubHandler<Arc<Session>>) -> Self {
+        Self {
+            handler,
+        }
+    }
+    pub fn start_ws(self, config: InformerConfig) -> Result<WsServer, String> {
+        let url = format!("{}:{}", config.interface, config.port);
+        let addr = url.parse().map_err(|_| format!("Invalid WebSockets listen host/port given: {}", url))?;
+        let server = self.handler;
+        let start_result = start_ws(&addr, server, config.max_connections);
+        match start_result {
+            Err(WsError::Io(ref err)) if err.kind() == io::ErrorKind::AddrInUse => {
+                Err(format!("WebSockets address {} is already in use, make sure that another instance of a Codechain node is not running or change the address using the --ws-port options.", addr))
+            },
+            Err(e) => {
+                Err(format!("WebSockets error: {:?}", e))
+            },
+            Ok(server) => {
+                cinfo!(INFORMER, "WebSockets Listening on {}", addr);
+                Ok(server)
+            },
+        }
+    }
+    pub fn event_subscription(&mut self, sender: Sender<Connection>) {
+        self.handler.add_subscription(
+            "Your event subscription begins",
+            ("register", move |params: Params, _, subscriber: Subscriber| {
+                let paramsclone = params.clone();
+                let unwraped_params: Result<Vec<String>, Error> = params.parse();
+                let allparams = unwraped_params.unwrap();
+                println!("####################");
+                println!("{}", allparams[0]);
+                if paramsclone == Params::None {
+                    subscriber
+                        .reject(Error {
+                            code: ErrorCode::ParseError,
+                            message: "Invalid parameters. Subscription rejected.".into(),
+                            data: None,
+                        })
+                        .unwrap();
+                    return
+                }
+                let mut rng = rand::thread_rng();
+                let sub_id = rng.gen();
+                println!("{}", sub_id);
+                let sink = subscriber.assign_id(SubscriptionId::Number(sub_id)).unwrap();
+                match sink.notify(Params::Array(vec![Value::String("You are now a new subscriber".into())])).wait() {
+                    Ok(_) => {}
+                    Err(_) => {
+                        cinfo!(INFORMER, "Subscription has ended, finishing.");
+                    }
+                }
```

Comment:
Is this needed?

Reply:
The purpose of this message is to let the client know that the subscription successfully occurs.
here is what the client receives after a successful subscription.
```
Received:  {"jsonrpc":"2.0","result":14308347713743523176,"id":1}
Received:  {"jsonrpc":"2.0","method":"Your event subscription begins","params":["You are now a new subscriber"]}
```